packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -271,17 +286,23 @@ export class Position {
         }
         return false;
     }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    toJSON(): any {
+        return { line: this.line, character: this.character };
+    }
 }
 
+@es5ClassCompat
```

Comment:
Why do we need this? Was the code broken before?

Reply:
Probably not. it wasn't broken. But (while I was having problems to build theia) I decided to make these classes to exact copies of according VS Code classes. The main changes here were absence of these `@es5ClassCompat` annotations and the absence of `toJSON(): any` methods (and some other methods).
But in theory I can revert these changes back. (Not sure if it would work, but it's easy to test)

Reply:
@akosyakov can you shed some light on this?

Reply:
It would be required if we down compile to es6, but code is still executed in es5 envs, then it is patching es6 classes to look like es5 classes. It is not interesting for us right now, since we still down compile to es5, but should not harm to copy.

Reply:
I have removed this change in the last version of the PR, so no changes in `types-impl.ts`